### ISSUE TITLE
About search-related results

### ISSUE THREAD

**Issue by Dmmm1997** [OpenSenseNova/SenseNova-MARS#10]
Great Work!!! I want to follow your work, but i meet some problem about the results.

When i evaluated the search-related benchmark (like mmserach, hr-mmsearch), the results doesn't match with the paper.

I want to ask what is the detailed setting for getting the results in the paper for the search-related benchmark.

My results are as follow (SenseNova-MARS-8B):

MMSearch: 53.8(paper 67.8)
HR-MMSearch: 33.1(paper 41.6)

I used the serper for evaluating the benchmark.

**Comment by yxchng** [comment]
@Dmmm1997 what is your search success rate? serper will return you url links and the code need to fetch the content of these links, how many of these fetching are successful?

**Comment by Dmmm1997** [comment]
> [@Dmmm1997](https://github.com/Dmmm1997) what is your search success rate? serper will return you url links and the code need to fetch the content of these links, how many of these fetching are successful?

(TaskRunner pid=393108) [VALIDATION] Batch 2/2 generation end - took 1004.12s
(TaskRunner pid=393108) 
(TaskRunner pid=393108) [VALIDATION] ===== Completed all 2 batches in 2726.95s =====
(TaskRunner pid=393108) Dumped generations to ./rollout_data/sensesearch_eval_SenseNova-MARS-8B_20260415_164555/validation/0.jsonl
(TaskRunner pid=393108) [VALIDATION] Judge LLM stats: total=278, successful=278, success_rate=100.00%
Training Progress: 0it [00:00, ?it/s]
(TaskRunner pid=393108) Initial validation metrics:
(TaskRunner pid=393108) {'val-aux/hr_mmsearch/llm_score/mean@1': 0.36065573770491804,
(TaskRunner pid=393108)  'val-aux/judge_llm_success_rate': 1.0,
(TaskRunner pid=393108)  'val-aux/judge_llm_successful': 278,
(TaskRunner pid=393108)  'val-aux/judge_llm_total': 278,
(TaskRunner pid=393108)  'val-aux/mat_search/llm_score/mean@1': 0.84,
(TaskRunner pid=393108)  'val-aux/num_turns/max': 18,
(TaskRunner pid=393108)  'val-aux/num_turns/mean': 7.520879120879121,
(TaskRunner pid=393108)  'val-aux/num_turns/min': 4,
(TaskRunner pid=393108)  'val-core/hr_mmsearch/reward/mean@1': 0.0,
(TaskRunner pid=393108)  'val-core/mat_search/reward/mean@1': 0.0,
(TaskRunner pid=393108)  'val-tool/image_search_tool/success_rate': 1.0,
(TaskRunner pid=393108)  'val-tool/image_search_tool/successful': 398,
(TaskRunner pid=393108)  'val-tool/image_search_tool/total': 398,
(TaskRunner pid=393108)  'val-tool/image_zoom_in_tool/success_rate': 1.0,
(TaskRunner pid=393108)  'val-tool/image_zoom_in_tool/successful': 212,
(TaskRunner pid=393108)  'val-tool/image_zoom_in_tool/total': 212,
(TaskRunner pid=393108)  'val-tool/num_tool_calls/max': 8,
(TaskRunner pid=393108)  'val-tool/num_tool_calls/mean': 2.7604395604395604,
(TaskRunner pid=393108)  'val-tool/num_tool_calls/min': 1,
(TaskRunner pid=393108)  'val-tool/success_rate': 1.0,
(TaskRunner pid=393108)  'val-tool/successful_calls': 1256,
(TaskRunner pid=393108)  'val-tool/text_search_tool/success_rate': 1.0,
(TaskRunner pid=393108)  'val-tool/text_search_tool/successful': 646,
(TaskRunner pid=393108)  'val-tool/text_search_tool/total': 646,
(TaskRunner pid=393108)  'val-tool/total_calls': 1256}
(TaskRunner pid=393108) step:0 - val-core/hr_mmsearch/reward/mean@1:0.0 - val-aux/hr_mmsearch/llm_score/mean@1:0.36065573770491804 - val-core/mat_search/reward/mean@1:0.0 - val-aux/mat_search/llm_score/mean@1:0.84 - val-aux/num_turns/min:4 - val-aux/num_turns/max:18 - val-aux/num_turns/mean:7.520879120879121 - val-tool/total_calls:1256 - val-tool/successful_calls:1256 - val-tool/success_rate:1.0 - val-tool/num_tool_calls/min:1 - val-tool/num_tool_calls/max:8 - val-tool/num_tool_calls/mean:2.7604395604395604 - val-tool/image_search_tool/total:398 - val-tool/image_search_tool/successful:398 - val-tool/image_search_tool/success_rate:1.0 - val-tool/text_search_tool/total:646 - val-tool/text_search_tool/successful:646 - val-tool/text_search_tool/success_rate:1.0 - val-tool/image_zoom_in_tool/total:212 - val-tool/image_zoom_in_tool/successful:212 - val-tool/image_zoom_in_tool/success_rate:1.0 - val-aux/judge_llm_total:278 - val-aux/judge_llm_successful:278 - val-aux/judge_llm_success_rate:1.0



This is the most recently results, using the same judge model and playwright for fetch the web information.

I find playwright will received higher performance compared with the JINA API.

**Comment by YFCodeDream** [comment]
Hi～, I used Jina API and get the similar results, just like the results you mentioned:
MMSearch: 53.8(paper 67.8)
HR-MMSearch: 33.1(paper 41.6)
but when I used playwright, there are so many search errors such as 403 forbidden, so I really wanna know how to reproduce the reported results :(
Could you give me some suggestions?

**Comment by yxchng** [comment]
@YFCodeDream The errors are expected if you are running from within China. Many sites are not surfable due to GFW. We use a server outside of China to run this work, so we do not have this problem. Our search success rate for playwright is about 80%. There will still be some pages that fail due to login required, javascript rendering, etc.

Jina is a good workaround for this problem but our current work did not use Jina. So you have to fine-tuned it on Jina results to get better results. Otherwise, if you really want to reproduce the results, try renting a server from Singapore or US.

**Comment by YFCodeDream** [comment]
Thank you very much for your continued patience in answering my questions :)
So my understanding is that we need to re-run the training pipeline for SenseNova-MARS, but during the training process, we should set it as the local knowledge server and do not pass the parameter "use_jina=true". Could you please explain what steps are needed for the operation you mentioned, "fine-tuning it based on Jina results"? Because we simply don't have the means to connect to foreign servers.
